src/FSharp.AWS.DynamoDB/Expression/ProjectionExpr.fs

```diff
@@ -2,158 +2,163 @@
 
         let ok, prop = ro.TryGetValue id.RootName
-        if ok then 
+
+        if ok then
             let cell = ref null
-            if aux cell id.NestedAttributes prop then value <- cell.Value; true
-            else false
-        else notFound id.NestedAttributes
 
-type ProjectionExpr = 
-    {
-        Attributes : AttributeId []
-        Ctor : Dictionary<string, AttributeValue> -> obj
-    }
+            if aux cell id.NestedAttributes prop then
+                value <- cell.Value
+                true
+            else
+                false
+        else
+            notFound id.NestedAttributes
+
+type ProjectionExpr =
+    { Attributes: AttributeId[]
+      Ctor: Dictionary<string, AttributeValue> -> obj }
 
-with
 
-    static member Extract (recordInfo : RecordTableInfo) (expr : Expr<'TRecord -> 'Tuple>) =
+    static member Extract (recordInfo: RecordTableInfo) (expr: Expr<'TRecord -> 'Tuple>) =
         let invalidExpr () = invalidArg "expr" "supplied expression is not a valid projection."
+
         match expr with
         | Lambda(r, body) when r.Type = recordInfo.Type ->
             let (|AttributeGet|_|) expr = QuotedAttribute.TryExtract (fun _ -> None) r recordInfo expr
 
             let (|Ignore|_|) e =
                 match e with
-                | Value(null, t) when t = typeof<unit> -> Some ()
-                | SpecificCall2 <@ ignore @> _ -> Some ()
+                | Value(null, t) when t = typeof<unit> -> Some()
+                | SpecificCall2 <@ ignore @> _ -> Some()
                 | _ -> None
 
             match body with
-            | Ignore -> 
+            | Ignore ->
                 let attr = AttributeId.FromKeySchema recordInfo.PrimaryKeySchema
-                { Attributes = [|attr|] ; Ctor = fun _ -> box () }
+                { Attributes = [| attr |]; Ctor = fun _ -> box () }
 
             | AttributeGet qa ->
                 let pickler = qa.Pickler
                 let attr = qa.Id
-                let ctor (ro : RestObject) =
+
+                let ctor (ro: RestObject) =
                     let mutable av = null
                     let ok = attr.View(ro, &av)
-                    if ok then pickler.UnPickleUntyped av
-                    else pickler.DefaultValueUntyped
 
-                { Attributes = [|attr|] ; Ctor = ctor }
+                    if ok then
+                        pickler.UnPickleUntyped av
+                    else
+                        pickler.DefaultValueUntyped
 
-            | NewTuple values -> 
-                let qAttrs = 
-                    values 
-                    |> Seq.map (function AttributeGet qa -> qa | _ -> invalidExpr ()) 
+                { Attributes = [| attr |]; Ctor = ctor }
+
+            | NewTuple values ->
+                let qAttrs =
+                    values
+                    |> Seq.map (function
+                        | AttributeGet qa -> qa
+                        | _ -> invalidExpr ())
```

Comment:
There doesn't appear to be a max expression length for `match`/`function`; it always breaks.

Reply:
maybe that's not the end of the world in this instance - it can work well for short one liner option matches, but I guess its also something that gets taken too far (speaking personally 😊 )